packages/components/src/globals/scss/vendor/@carbon/elements/scss/type/_styles.scss

```diff
@@ -671,3 +672,21 @@ $tokens: (
     @include properties(map-remove($token, 'breakpoints'));
   }
 }
+
+/// Type scole that maps font size to icon size
+/// @type Map
+/// @access public
+/// @group @carbon/type
+$carbon--type-icon-scale: (
+  0.875rem: rem(16px),
+  1rem: rem(20px),
+);
+
+/// Translate font size to icon size
+/// @param {Number} $font-size - Font size in rems
+/// @return {Number} Icon size in rems
+/// @access public
+/// @group @carbon/type
+@function carbon--type-icon-size($font-size) {
```

Comment:
The intent behind these additions seem great, should definitely bring them upstream. Are there any other additions that'd make working with stuff in this PR easier?

Reply:
I'm loving the idea here as well. While this is the general rule we're following, there are a couple of components like the code snippet family where the icon doesn't change its size to `20px`.

Reply:
@photodow I think the rule we want to codify is that the icon size matches the type size, and is contained with the set `[16, 20, 24, 32]`

Reply:
My thinking was only components that had a clear type/icon pairing (e.g. if they are inline) would use the new `carbon--type-icon-size()`. If a component uses a standalone icon (no inline text), then the component shouldn't use this function.